internal/js_library/v2/babel.js

```diff
@@ -0,0 +1,45 @@
+// A simple wrapper around babel-core due to https://github.com/babel/babel/issues/8193
+
+const {writeFile} = require('fs');
+const path = require('path');
+
+const babel = require('@babel/core');
+const mkdirp = require('mkdirp');
+const program = require('commander');
+
+program.version('1.0.0')
+    .usage('[options] <files...>')
+    .option('-o, --out-dir <value>', 'Output directory for created files')
+    .option('-c, --config-file <value>', 'babel.rc.js config file')
+    .parse(process.argv);
+
+const outDir = program['outDir'] || '';
+
+let babelConfig = {};
+if (program['configFile']) {
+  // Note: We do not want to use bazel resolve mechanisms, so use path resolve to get the absolute
+  // path and load that.
+  babelConfig = require(path.resolve(program['configFile']));
+}
+
+for (let i = 0; i < program.args.length; i += 1) {
+  const input = program.args[i];
+  const output = path.join(outDir, path.basename(input));
```

Comment:
this is the source of the issue around nested files. We should pass into this binary directly the output paths for each input, and they should be based on the relative path to the package for each file.